File: Models/MovieSearchResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Models
{
    public class MovieSearchResult
    {
        [JsonPropertyName("results")]
        public List<Movie> Results { get; set; }
        [JsonPropertyName("page")]
        public int Page { get; set; }
        [JsonPropertyName("total_pages")]
        public int TotalPages { get; set; }
    }
}

File: frontend/moviedatabase/lib/Models/MovieSearchResult.dart
import 'Movie.dart';

class MovieSearchResult {
  final int page;
  final List<Movie> results;
  final int totalResults;
  final int totalPages;

  MovieSearchResult({
    required this.page,
    required this.results,
    required this.totalResults,
    required this.totalPages,
  });

  factory MovieSearchResult.fromJson(Map<String, dynamic> json) {
    var list = json['results'] as List? ?? [];
    List<Movie> movieList = list.map((i) => Movie.fromJson(i)).toList();

    return MovieSearchResult(
      page: json['page'] ?? 1,
      results: movieList,
      totalResults: json['total_results'] ?? json['totalResults'] ?? 0,
      totalPages: json['total_pages'] ?? json['totalPages'] ?? 1,
    );
  }
}
